crates/flox-rust-sdk/src/providers/git.rs

```diff
@@ -432,15 +445,80 @@ impl GitProvider for GitCommandProvider {
         let _out = GitCommandProvider::run_command(
             GitCommandProvider::new_command(&self.workdir)
                 .arg("branch")
-                .arg(branch)
-                .arg("--set-upstream")
+                .arg("--set-upstream-to")
                 .arg(format!("{origin_name}/{branch}")),
         )
         .await?;
 
         Ok(())
     }
 
+    /// Retrieve information about the remot origin for the current branch/repo
+    ///
+    /// Return a tuple containing
+    ///
+    /// 1. the remote name of the current branch (or "origin" if no upstream configured)
+    /// 2. the remote url
+    /// 3. (if configured) a tuple containing
+    ///    1. the upstream branch name
+    ///    2. the current revision of the branch
+    async fn get_origin(
+        &self,
+    ) -> Result<(String, String, Option<(String, String)>), Self::GetOriginError> {
+        let (remote_name, remote_branch) = {
+            let reference = GitCommandProvider::run_command(
+                GitCommandProvider::new_command(&self.workdir)
+                    .arg("rev-parse")
+                    .arg("--abbrev-ref")
+                    .arg("--symbolic-full-name")
+                    .arg("@{u}"),
+            )
+            .await;
+
+            match reference {
+                Err(_) => {
+                    error!("Couldn't determine upstream remote name for the current branch, defaulting to 'origin'");
+                    ("origin".to_string(), None)
+                },
+                Ok(reference) => reference
+                    .to_string_lossy()
+                    .trim()
+                    .split_once('/')
+                    .map(|(name, branch)| (name.to_string(), Some(branch.to_string())))
+                    .unwrap(),
+            }
+        };
+
+        let url = GitCommandProvider::run_command(
+            GitCommandProvider::new_command(&self.workdir)
+                .arg("remote")
+                .arg("get-url")
+                .arg(&remote_name),
+        )
+        .await?
+        .to_string_lossy()
+        .trim()
+        .to_string();
+
+        let branch_and_commit = match remote_branch {
+            Some(branch) => Some((
+                branch.clone(),
```

Comment:
Lol had to do a double take, my brain saw branch and clone and thought this was doing something git related 😆